reader.go

```diff
@@ -0,0 +1,211 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/fuzxxl/nfc/2.0/nfc"
+	"golang.org/x/net/context"
+)
+
+// DefaultModulation gives defaults for the modulation type and Baudrate.
+// Currently, only nfc.ISO14443a is supported for the type. If the default
+// BaudRate is not supported by the reader, the fallback is the lowest
+// supported value.
+var DefaultModulation = nfc.Modulation{
+	Type:     nfc.ISO14443a,
+	BaudRate: nfc.Nbr106,
+}
+
+// PollingInterval gives the interval of polling for new cards.
+var PollingInterval = 100 * time.Millisecond
+
+// Reader abstracts the NFC Reader functions we need, for testing. A Reader
+// must be safe for concurrent use.
+type Reader interface {
+	// GetNextUID blocks until a new card is put on the reader and returns the
+	// UID of the card.
+	GetNextUID() (uid []byte, err error)
+	// Close closes the Reader and cleans up any resources held.
+	Close() error
+}
+
+type nfcReader struct {
+	nfc.Device
+	m nfc.Modulation
+	// uids is used to pass card-uids between goroutines
+	uids chan []byte
+	// errs passes errors from polling
+	errs chan error
+	// used for threadsafe cancellation
+	ctx    context.Context
```

Comment:
Das ist i.A. wahr, aber in diesem Fall missbrauche ich das ein bisschen *nur* für threadsafe cancellation. Die einzige Alternative dazu, die mir einfiele, wäre, die entsprechende Funktionalität aus net/context zu kopieren. Das fand ich ein bisschen umständlich, nur für diese Funktionalität.